WcaOnRails/app/webpacker/components/CompetitionOverview/CompetitionTable.js

```diff
@@ -0,0 +1,178 @@
+import React from 'react';
+import {
+  List, Icon, Popup, Loader,
+} from 'semantic-ui-react';
+
+import I18n from '../../lib/i18n';
+import { dayDifferenceFromToday, PseudoLinkMarkdown } from '../../lib/utils/competition-table';
+
+function CompetitionTable({
+  competitionData,
+  title,
+  shouldShowRegStatus,
+  shouldIncludeCancelled,
+  selectedEvents,
+  isLoading,
+  hasMoreCompsToLoad,
+  isSortedByAnnouncement = false,
+  isRenderedAboveAnotherTable = false,
+}) {
+  const competitions = competitionData?.filter((comp) => (
+    (!comp.cancelled_at || shouldIncludeCancelled)
+    && (selectedEvents.every((event) => comp.event_ids.includes(event)))
+  ));
+
+  return (
+    <List divided relaxed>
+      <List.Item>
+        <strong>
+          {`${title} (${competitions ? competitions.length : 0}${hasMoreCompsToLoad ? '...' : ''})`}
+        </strong>
+      </List.Item>
+      {competitions?.map((comp, index) => (
+        <React.Fragment key={comp.id}>
+          <ConditionalYearHeader
+            competitions={competitions}
+            index={index}
+            isSortedByAnnouncement={isSortedByAnnouncement}
+          />
+          <List.Item className={`${comp.isProbablyOver ? ' past' : ' not-past'}${comp.cancelled_at ? ' cancelled' : ''}`}>
+            <span className="date">
+              <StatusIcon
+                comp={comp}
+                shouldShowRegStatus={shouldShowRegStatus}
+                isSortedByAnnouncement={isSortedByAnnouncement}
+              />
+              {comp.dateRange}
+            </span>
+            <span className="competition-info">
+              <div className="competition-link">
+                <span className={` fi fi-${comp.country_iso2}`} />
+                &nbsp;
+                <a href={comp.url}>{comp.displayName}</a>
+              </div>
+              <div className="location">
+                <strong>{comp.countryName}</strong>
+                {`, ${comp.cityName}`}
+              </div>
+              <div className="venue-link">
+                <PseudoLinkMarkdown text={comp.venue} />
+              </div>
+            </span>
+          </List.Item>
+        </React.Fragment>
+      ))}
+      {/* Could not figure out why Semantic UI's animated loader icon doesn't show */}
+      {
+        isLoading
+        && (
+          <List.Item style={{ textAlign: 'center' }}>
+            <Loader active inline="centered" size="small">
+              {I18n.t('competitions.index.loading_comps')}
+            </Loader>
+          </List.Item>
+        )
+      }
+      {
+        !hasMoreCompsToLoad
+        && !isLoading
+        && !isRenderedAboveAnotherTable
+        && <EndOfCompListMessage numCompetitions={competitions?.length} />
+      }
```

Comment:
It seems to me like there should be a single footer or end-of-list component that's responsible for showing the message, the bottom-ref div, or nothing. (It could also include the loading spinner maybe?) Rather than having this component (sometimes) show the message, and the parent (sometimes) show the ref.

Does this message need to be in the `List`, or could it be a separate component outside (after) the `List`?

Reply:
They don't have to be in the list, the font and size would just look more alike and the spacing is more even. 

A footer does sound like a clean way to organize these things though.